src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -524,25 +525,28 @@ private bool TryGetUnmanagedCallingConventionFromModOpt(MethodSignature signatur
                 if (defType.Namespace != "System.Runtime.CompilerServices")
                     continue;
 
-                // Take the first recognized calling convention in metadata.
-                switch (defType.Name)
+                // Look for a recognized calling convention in metadata.
+                CorInfoCallConv? callConvLocal = defType.Name switch
                 {
-                    case "CallConvCdecl":
-                        callConv = CorInfoCallConv.CORINFO_CALLCONV_C;
-                        return true;
-                    case "CallConvStdcall":
-                        callConv = CorInfoCallConv.CORINFO_CALLCONV_STDCALL;
-                        return true;
-                    case "CallConvFastcall":
-                        callConv = CorInfoCallConv.CORINFO_CALLCONV_FASTCALL;
-                        return true;
-                    case "CallConvThiscall":
-                        callConv = CorInfoCallConv.CORINFO_CALLCONV_THISCALL;
-                        return true;
+                    "CallConvCdecl"     => CorInfoCallConv.CORINFO_CALLCONV_C,
+                    "CallConvStdcall"   => CorInfoCallConv.CORINFO_CALLCONV_STDCALL,
+                    "CallConvFastcall"  => CorInfoCallConv.CORINFO_CALLCONV_FASTCALL,
+                    "CallConvThiscall"  => CorInfoCallConv.CORINFO_CALLCONV_THISCALL,
+                    _ => null
+                };
+
+                if (callConvLocal.HasValue)
+                {
+                    // Error if there are multiple recognized calling conventions
+                    if (found)
+                        ThrowHelper.ThrowInvalidProgramException(ExceptionStringID.InvalidProgramMultipleCallConv, MethodBeingCompiled);
```

Comment:
Nit: might be better to report this same as the VM side, without the method responsible. We don't actually know if we're blaming the right method. We could be inlining.